bazel-mode.el

```diff
@@ -0,0 +1,131 @@
+;;; bazel-mode.el --- Emacs major mode for editing Bazel BUILD and WORKSPACE files -*- lexical-binding: t; -*-
+
+
+;; Copyright (C) 2018 Google LLC
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;      http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+;;; Commentary:
+;;
+;; This package provides Emacs bazel-mode which performs syntax
+;; highlighting for Bazel WORKSPACE and BUILD files.
+;;
+;;; Code:
+
+;;;###autoload
+(define-derived-mode bazel-mode prog-mode "Bazel"
+  "Major mode for editing Bazel BUILD and WORKSPACE files."
+  (setq font-lock-defaults '((bazel-font-lock-keywords))))
+
+(defvar bazel-mode-syntax-table
```

Comment:
Typically you should define the syntax table variable before the mode because `define-derived-mode` differs whether the variable has already been defined or not. See https://www.gnu.org/software/emacs/manual/html_node/elisp/Derived-Modes.html.